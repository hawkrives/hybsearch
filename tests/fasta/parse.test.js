/* eslint-env jest */
'use strict'

const expect = require('expect')
const { parseFasta } = require('../../server/lib/fasta/parse')

describe('fasta/parse', () => {
	it('should handle basic files', () => {
		const actual = parseFasta(`
> Emydura_subglobosa__KC755190
ggaacaataaattatcacctcaaaagacac

> Emydura_victoriae__KC755189
ggaacaatcaattattaccccacaaagac

> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
  `)

		expect(actual).toMatchSnapshot()
	})

	it('should handle multi-line sequences', () => {
		const actual = parseFasta(`
> Emydura_subglobosa__KC755190
ggaacaataaattatcacctcaaaagacac
ggaacaataaattatcacctcaaaagacac
ggaacaataaattatcacctcaaaagacac

> Emydura_victoriae__KC755189
ggaacaatcaattattaccccacaaagac
ggaacaatcaattattaccccacaaagac
ggaacaatcaattattaccccacaaagac
ggaacaatcaattattaccccacaaagac
ggaacaatcaattattaccccacaaagac
ggaacaatcaattattaccccacaaagac

> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
ggaacaatcaattattaccccacaaagac
ggaacaatcaattattaccccacaaagac
  `)

		expect(actual).toMatchSnapshot()
	})
})
