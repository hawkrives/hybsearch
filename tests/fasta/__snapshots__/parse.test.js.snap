// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fasta/parse should handle basic files 1`] = `
Array [
  Object {
    "sequence": "ggaacaataaattatcacctcaaaagacac",
    "species": "Emydura_subglobosa__KC755190",
  },
  Object {
    "sequence": "ggaacaatcaattattaccccacaaagac",
    "species": "Emydura_victoriae__KC755189",
  },
  Object {
    "sequence": "ggaacaatcaattattaccccacaaggaa",
    "species": "Emydura_victoriae__KC755188",
  },
]
`;

exports[`fasta/parse should handle multi-line sequences 1`] = `
Array [
  Object {
    "sequence": "ggaacaataaattatcacctcaaaagacacggaacaataaattatcacctcaaaagacacggaacaataaattatcacctcaaaagacac",
    "species": "Emydura_subglobosa__KC755190",
  },
  Object {
    "sequence": "ggaacaatcaattattaccccacaaagacggaacaatcaattattaccccacaaagacggaacaatcaattattaccccacaaagacggaacaatcaattattaccccacaaagacggaacaatcaattattaccccacaaagacggaacaatcaattattaccccacaaagac",
    "species": "Emydura_victoriae__KC755189",
  },
  Object {
    "sequence": "ggaacaatcaattattaccccacaaggaaggaacaatcaattattaccccacaaagacggaacaatcaattattaccccacaaagac",
    "species": "Emydura_victoriae__KC755188",
  },
]
`;
